vault/core.go

```diff
@@ -1234,10 +1246,10 @@ func (c *Core) unsealPart(ctx context.Context, seal Seal, key []byte, useRecover
 	} else {
 		masterKey = recoveredKey
 	}
-	newRecoveryKey := masterKey
 
-	// If we have a migration seal, now's the time!
-	if c.migrationSeal != nil {
+	if c.migrationInfo != nil {
+		c.migrationInfo.newRecoveryKey = make([]byte, len(masterKey))
```

Comment:
Although this works as is, I think it would be clearer if you used `recoveredKey` instead of `masterKey` here.

Reply:
Done